python/tank/templatekey.py

```diff
@@ -256,6 +273,145 @@ def _as_string(self, value):
         return value if isinstance(value, basestring) else str(value)
 
 
+class TimestampKey(TemplateKey):
+    """
+    Key whose value is a time string formatted with strftime.
+    """
+
+    def __init__(
+        self,
+        name,
+        utc_default=False,
+        format_spec="%Y-%m-%d-%H-%M-%S"
+    ):
+        """
+        Constructor
+        :param name: Name by which the key will be refered.
+        :param utc_default: If True, the default value will be an UTC time. Defaults to False.
```

Comment:
how is this expected to be used by clients/by us? I think maybe a better way to handle this is to have two timestamp types available to users - one `type: timestamp` and one `type: utctimestamp`. What do you think? This could then be represented as two classes here. Maybe its worth just doing the local time for now and wait with the UTC? Or were both requested?